tests/json/generate_parsing_test.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+
+import glob
+import os
+import os.path
+import sys
+
+
+def transform_string_sub(dest_file, iterator):
+    for c in iterator:
+        if c < ord(b' ') or c >= 0x7f or c == ord(b'"') or c == ord(b'\\'):
+            dest_file.write(b'\\%03o' % (c,))
+        else:
+            dest_file.write(b'%c' % (c,))
+
+
+def iter_file_as_ints(input_file):
+    while True:
+        c = input_file.read(1)
+        if not c:
+            break
+        yield ord(c)
+
+
+def transform_string(dest_file, input_file, test_name, esc_test_name):
+    dest_file.write(b'static const char ')
+    dest_file.write(esc_test_name)
+    dest_file.write(b'[] = "')
+    transform_string_sub(dest_file, iter_file_as_ints(input_file))
+    dest_file.write(b'";\n')
+
+    dest_file.write(b'static const char ')
+    dest_file.write(esc_test_name)
+    dest_file.write(b'_name[] = "')
+    transform_string_sub(dest_file, test_name)
+    dest_file.write(b'";\n')
+
+    dest_file.write(b'\n')
+
+
+def escape_test_name(test_name, index):
+    escaped_str = b''.join(
+        b'%c' % (c,) if c in range(ord('a'), ord('z') + 1) or
+        c in range(ord('A'), ord('Z') + 1) or
+        c in range(ord('0'), ord('9') + 1) and index > 0 else
+        b'_%02x' % (c,)
+        for index, c in enumerate(test_name)
+    )
+    return b'json_parsing_test_%s_%d' % (escaped_str, index)
+
+
+def main(dest_path='./generated-parsing-tests.c',
+         source_path='./JSONTestSuite',
+         cwd=os.path.dirname(os.path.realpath(__file__))):
+    test_names = []
+    cwd = os.path.abspath(cwd)
+    with open(os.path.join(cwd, dest_path), 'wb') as dest_file:
+        # License:
+        dest_file.write(b'/* This file was generated using the tests in '
+                        b'JSON Parsing Test Suite,\n')
+        dest_file.write(b' * which has the following license information:\n')
+        dest_file.write(b'\n')
+        with open(os.path.join(cwd, source_path, './LICENSE'), 'rb') as license:
+            dest_file.write(license.read())
+        dest_file.write(b'\n')
+        dest_file.write(b' */\n')
+        dest_file.write(b'\n')
+        dest_file.write(b'/* Upstream version: ???????? */\n')
+        dest_file.write(b'#error "PLEASE EDIT UPSTEAM VERSION HASH MANUALLY AND REMOVE THIS LINE!"\n')
+        dest_file.write(b'\n')
+
+        # Includes etc.:
+        dest_file.write(b'#include "tests-json.h"\n')
+        dest_file.write(b'\n')
+        dest_file.write(b'#include <stddef.h>\n')
+        dest_file.write(b'\n')
+        dest_file.write(b'#pragma GCC diagnostic ignored '
+                        b'"-Woverlength-strings"\n')
+        dest_file.write(b'\n')
+
+        # Strings:
```

Comment:
Maybe name this function `generate_fixtures()` or something like that, to make clearer, that those strings are actually the test fixtures for the tests, not a test suite.